data/users.js

```diff
@@ -30,17 +31,20 @@ module.exports = {
         // Check if username already exists
         const user = await this.getUser(username);
 
-        if (user != null)
-        {
-            throw 'Cannot create username since it already exists!'
-        }
+        if (user != null) throw 'Cannot create username since it already exists!';
 
-        // @todo check if email matches a university's email domain
+        //get Email domain
+        email = email.trim();
+        email = email.split("@")[1];
```

Comment:
Careful, this is overriding the email variable which is used below to create the user